src/Swarm/Game/Step.hs

```diff
@@ -1230,6 +1385,34 @@ execConst c vs s k = do
       _ -> badConst
     Grab -> doGrab Grab'
     Harvest -> doGrab Harvest'
+    Ignite -> case vs of
+      [VDir d] -> do
+        let verb = "ignite"
+            verbed = "ignited"
+
+        (loc, me) <- lookInDirection d
+        -- Ensure there is an entity here.
+        e <-
+          me `isJustOrFail` ["There is nothing here to", verb <> "."]
```

Comment:
Don't we have code very similar to this in other places?  Is there an opportunity to factor out some repeated code?